pallets/clients-info/src/lib.rs

```diff
@@ -108,22 +129,73 @@ pub mod pallet {
 			client_name: NameOf<T>,
 			release: ClientRelease<UriOf<T>, T::Hash>,
 		) -> DispatchResult {
-			ensure_root(origin)?;
+			Pallet::<T>::check_origin_rights(origin)?;
 			PendingClientReleases::<T>::insert(client_name, release.clone());
 			Self::deposit_event(Event::<T>::NotifyClientRelease { release });
 			Ok(())
 		}
+
+		#[pallet::call_index(2)]
+		#[pallet::weight(<T as Config>::WeightInfo::authorize_account())]
+		#[transactional]
+		pub fn authorize_account(origin: OriginFor<T>, account_id: T::AccountId) -> DispatchResult {
+			Pallet::<T>::check_origin_rights(origin)?;
+
+			if !<AuthorizedAccounts<T>>::contains_key(&account_id) {
+				Self::deposit_event(Event::<T>::AccountIdAuthorized(account_id.clone()));
+				<AuthorizedAccounts<T>>::insert(account_id, ());
+			}
+
+			Ok(())
+		}
+
+		#[pallet::call_index(3)]
+		#[pallet::weight(<T as Config>::WeightInfo::deauthorize_account())]
+		#[transactional]
+		pub fn deauthorize_account(
+			origin: OriginFor<T>,
+			account_id: T::AccountId,
+		) -> DispatchResult {
+			if let Ok(origin_account_id) = ensure_signed(origin.clone()) {
+				Pallet::<T>::check_non_root_rights(origin.clone())?;
+				ensure!(
+					account_id != origin_account_id,
+					Error::<T>::UserUnableToDeauthorizeThemself
+				);
+
+				// do not deauthorize the root account
+				let account_id_to_origin =
+					OriginFor::<T>::from(RawOrigin::Signed(account_id.clone()));
+				ensure!(
+					ensure_root(account_id_to_origin).is_err(),
+					Error::<T>::UserUnableToDeauthorizeRootAccount
+				);
```

Comment:
I think we don't really have to handle this case since the implementation of `check_origin_rights` does not even care if `root` is part of the list of the `AuthorizedAccounts`. So, while it does not make sense to add `root` to the list of authorized accounts (since it can call all functions anyways), it is not a problem if the user of this pallet adds it to the authorized accounts.
Or am I seeing this wrong @b-yap?

Reply:
case removed